app/src/main/java/ca/pkay/rcloneexplorer/RemoteConfig/DynamicConfig.kt

```diff
@@ -0,0 +1,271 @@
+
+            val layout = LinearLayout(mContext)
+            layout.orientation = LinearLayout.VERTICAL
+
+            val textViewTitle = TextView(mContext)
+            textViewTitle.contentDescription = it.name
+            textViewTitle.text = it.name
+            textViewTitle.typeface = Typeface.DEFAULT_BOLD
+            layout.addView(textViewTitle)
+
+
+            val textViewDescription = TextView(mContext)
+            textViewDescription.contentDescription = it.help
+            textViewDescription.text = it.help
+            layout.addView(textViewDescription)
+
+
+            Log.e("TAG", "      Opt: ${it.name}")
+            Log.e("TAG", "      Opt: ${it.type}")
+            when (it.type) {
+                "string" -> {
+                    if(it.isPassword) {
+                        val input = getAttachedEditText(it.name, layout)
+                        input.inputType =
+                            InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+
+                        setTextInputListener(input, it.name)
+
+                    } else if(it.examples.size > 0) {
+                        createSpinnerFromExample(it, it.name ,layout)
+                    } else {
+                        val input = getAttachedEditText(it.name, layout)
+                        setTextInputListener(input, it.name)
+                    }
+
+
+                }
+                else -> {
+                    val unknownType = getAttachedEditText(it.name, layout)
+                    //unknownType.hint = it.type
+                    setTextInputListener(unknownType, it.name)
+                }
+            }
+
+
+            //val divider = View(mContext)
+            //divider.setBackgroundColor(resources.getColor(android.R.color.darker_gray))
+            //val lp = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, getDPasPixel(1).toInt())
+            //divider.layoutParams = lp
+            //layout.addView(divider)
+
+
+            val cardContainer = getCard()
+            cardContainer.addView(layout)
+            mFormView?.addView(cardContainer)
+        }
+    }
+
+    private fun getDPasPixel(dp: Int): Float {
+        val scale = resources.displayMetrics.density
+        return (dp * scale + 0.5f)
+    }
+
+
+    private fun createSpinnerFromExample(option: ProviderOption, hint: String, layout: LinearLayout): View {
+
+
+        val padding = resources.getDimensionPixelOffset(R.dimen.cardPadding)
+        val textinput = TextInputLayout(ContextThemeWrapper(activity, R.style.Widget_MaterialComponents_TextInputLayout_OutlinedBox_ExposedDropdownMenu))
+        textinput.hint = hint
+        textinput.boxBackgroundMode = TextInputLayout.BOX_BACKGROUND_OUTLINE
+        textinput.setPadding(0, padding, 0, 0)
+
+        var input = AutoCompleteTextView(textinput.context)
+        input.setPadding(padding)
+        val items = ArrayList<String>()
+
+        option.examples.forEach { example ->
```

Comment:
This is exactly the kind of stuff where i wil slip with this, because i cannot extensively test every remote.

Thanks for the writeup, i will have to read this thoroughly to make this work properly, especially to prevent weird behaviour with rclone